edd_utils/__init__.py

```diff
@@ -34,18 +43,19 @@ def login(edd_server='edd.jbei.org', user=None):
         if os.path.exists(rc):
             config = RawConfigParser()
             config.read(rc)
-            # TODO: Check if edd_server section exists
-            user = config[edd_server]['username']
-            password = config[edd_server]['password']
+            if edd_server in config:
+                user = config[edd_server].get('username')
+                password = config[edd_server].get('password')
 
-        # If not, get username / password from input
-        else:
+    # prompt user for username if config didn't exist or lacked a username
+    while not user:
+        if default_to_system_user:
             user = getpass.getuser()
+        else:
+            user = input(f"Username for {edd_server}: ")
+            user = user.strip()
```

Comment:
Shouldn't this be outside of the loop, in case 'user' is provided as an argument?

Reply:
I don't think so, but it also wouldn't hurt.  If the client (a programmer) provides a username via code, I think it's reasonable to just use that value.  In this particular code block, we're prompting for direct user input (e.g. maybe from someone who isn't a programmer using a pre-prepared notebook).  In direct user input, additional spaces may not be quite as obvious as in a string literal surrounded by quotes.

Reply:
Good point.